miden-lib/asm/kernels/transaction/lib/account.masm

```diff
@@ -365,32 +333,35 @@ end
 #! Output: [offset_slot_index]
 export.apply_storage_offset
     # verify that slot_index is in bounds
-    dup.1 push.MAX_STORAGE_SLOT_INDEX lt assert.err=ERR_STORAGE_SLOT_INDEX_OUT_OF_BOUNDS
+    dup.1 exec.get_max_num_storage_slots lt assert.err=ERR_STORAGE_SLOT_INDEX_OUT_OF_BOUNDS
     # => [storage_offset, slot_index]
 
     # verify that the computed offset_slot_index is in bounds
-    add dup push.MAX_STORAGE_SLOT_INDEX lt assert.err=ERR_STORAGE_SLOT_INDEX_OUT_OF_BOUNDS
+    add dup exec.get_max_num_storage_slots lt assert.err=ERR_STORAGE_SLOT_INDEX_OUT_OF_BOUNDS
     # => [offset_slot_index]
 end
 
-#! Gets an item from the account storage. Panics if the index is out of bounds.
+#! Gets an item from the account storage
+#!
+#! Panics:
+#!    - index is out of bounds.
 #!
 #! Stack: [index]
 #! Output: [VALUE]
 #!
 #! - index is the index of the item to get.
 #! - VALUE is the value of the item.
 export.get_item
-    # get the storage root
-    exec.memory::get_acct_storage_root
-    # => [STORAGE_ROOT, index]
+    # check that index is in bounds
+    dup exec.get_max_num_storage_slots lt assert.err=ERR_STORAGE_SLOT_INDEX_OUT_OF_BOUNDS
+    # => [index, V']
```

Comment:
Reference to `V'` in the comment does not seem right.

But also, don't we usually call this method after `apply_storage_offset` is called? If so, do we need this check?

Reply:
I agree with you, for now we do only call this procedure after `apply_storage_offset` hence the check is done twice, but in the future we are not sure that we will always call this function always after `apply_storage_offset`. 

I have the feeling that it makes sense to have the checks at the kernel level to make sure that out of bound accesses never occur.

Reply:
Hence should we keep the check:
- Only at the kernel level
- Only at the API level
- At both levels

Reply:
I would probably keep these checks (as much as possible) in the `api.mams` file. This will also allow us to use procedures in the kernel modules directly (e.g., if we want to read storage from the context of doing something else - i.e., not from user procedure invocation).

Reply:
Understood, nonetheless if we remove the bound check from the kernel procedures they will differ from the Rust implementation, I am not entirely sure how important it is to keep them in sync. If calling `get_item` from the kernel then they won't have the same behaviour, but when calling `get_account_item` from the api then the behaviour of the Rust and MASM should be close.

Reply:
Made the requested change.